File: WEB programming/programming-www/w4code14/process_names.rb
# process_names.rb - A simple Ruby program to
#  illustrate the use of arrays
#  Input: A list of lines of text, where each line
#         is a person's name
# Output: The input names, after all letters are 
#         converted to uppercase, in alphabetical order

index = 0
names = Array.new

# Loop to read the names and process them
while(name = gets)

# Convert the name's letters to uppercase and put it
#  in the names array
  names[index] = name.chomp.upcase
  index += 1
end

# Sort the array in place and display it
names.sort!
puts "The sorted array"
for name in names
  puts name
end

File: WEB programming/programming-www/w4code8/process_names.pl
# process_names.pl - A simple program to illustrate
#                     the use of arrays
#  Input: A file, specified on the command line, of 
#         lines of text, where each line is a person’s
#         name
# Output: The input names, after all letters are converted
#         to uppercase, in alphabetical order 

$index = 0;

#>>> Loop to read the names and process them
while($name = <>) {

#>>> Convert the name’s letters to uppercase and put it in
#>>>  the names array
    $names[$index++] = uc($name);
}

#>>> Display the sorted list of names
print "\nThe sorted list of names is:\n\n\n";
foreach $name (sort @names) {
    print ("$name \n");
}
